File: server/internal/expenses/payment.go
package expenses

import (
	"time"

	"gorm.io/gorm"
)

type Payment struct {
	ID        uint           `json:"id" gorm:"primaryKey;type:bigint"`
	WalletID  uint           `json:"wallet_id" gorm:"type:bigint;not null;index"`
	Amount    float64        `json:"amount" gorm:"type:numeric(12,2);not null"`
	Date      time.Time      `json:"date" gorm:"type:date;not null;index"`
	Note      string         `json:"note" gorm:"type:text"`
	UserID    uint           `json:"user_id" gorm:"type:bigint;not null;index"`
	CreatedAt time.Time      `json:"created_at"`
	UpdatedAt time.Time      `json:"updated_at"`
	DeletedAt gorm.DeletedAt `json:"-" gorm:"index"`

	Wallet   Wallet                  `json:"wallet,omitempty" gorm:"foreignKey:WalletID;constraint:OnUpdate:CASCADE,OnDelete:CASCADE;"`
	Expenses []PaymentExpenseSummary `json:"expenses,omitempty" gorm:"-"`
}

type PaymentExpenseTypeSummary struct {
	ID    uint   `json:"id"`
	Name  string `json:"name"`
	Icon  string `json:"icon"`
	Color string `json:"color"`
}

type PaymentExpenseSummary struct {
	ID            uint                      `json:"id"`
	PaymentID     uint                      `json:"payment_id"`
	ExpenseTypeID uint                      `json:"expense_type_id"`
	Amount        float64                   `json:"amount"`
	ExpenseType   PaymentExpenseTypeSummary `json:"expense_type"`
}

File: client/src/types/payment.ts
export interface PaymentWalletSummary {
  id: number
  name: string
  icon: string
  color: string
  is_credit: boolean
  is_cash: boolean
}

export interface PaymentExpenseTypeSummary {
  id: number
  name: string
  icon: string
  color: string
}

export interface PaymentExpenseSummary {
  id: number
  payment_id: number
  expense_type_id: number
  amount: number
  expense_type: PaymentExpenseTypeSummary
}

export interface Payment {
  id: number
  wallet_id: number
  amount: number
  date: string
  note: string
  user_id: number
  created_at: string
  updated_at: string
  wallet?: PaymentWalletSummary | null
  expenses?: PaymentExpenseSummary[]
}

export interface CreatePaymentRequest {
  wallet_id: number
  amount: number
  date: string
  note?: string
  expense_ids?: number[]
  auto_create_default_expense?: boolean
}

export type UpdatePaymentRequest = CreatePaymentRequest

export interface PaymentListResponse {
  payments: Payment[]
  total: number
}

export interface PaymentMonthlyTotalResponse {
  from: string
  to: string
  total: number
}